src/main/java/htsjdk/samtools/reference/AbstractFastaSequenceFile.java

```diff
@@ -127,7 +127,7 @@ protected String getAbsolutePath() {
 
```

Comment:
Get absolute path needs to change because it will crash with NPE if there's no path.  It should crash with a more useful exception instead.

Reply:
There are a lot of places that use `getAbsolutePath` as part of error messages.  Those should be changed to use the new name field that I mention below, or have explicit checks for the availability of path first.  It's not good to crash when generating an error message...

Reply:
Fixed to use source, which defaults to the absolute path if a path is supplied.